thunder/dynamo/utils.py

```diff
@@ -304,6 +331,14 @@ def is_node_supported_by_thunder(node: torch.fx.Node) -> tuple[bool, SplitReason
         )
         return False, split_reason
 
+    # If the operation is higher order function, check whether the submodule is supported by Thunder
```

Comment:
I think we should rewrite `If the operation is higher order function` -> `If the operation is higher order function for checkpointing`, as I am not sure this will work with other operations in higher order ops.